geopandas/tools/overlay.py

```diff
@@ -136,13 +137,13 @@ def overlay(df1, df2, how, use_sindex=True):
         prop1 = None
         prop2 = None
         for cand_id in candidates1:
-            cand = df1.ix[cand_id]
+            cand = df1.loc[cand_id]
```

Comment:
this one is not necessarily correct, if `cand_id` is an integer position and `df1` has another index

Reply:
hmm, but I assume it are the actual index labels, as it is doing `i for i, x in df2.iterrows()` above

Reply:
I think this behavior is correct (I looked a bit at this before choosing `loc` over `iloc`). However, better to add tests of behavior with a non-integer index before moving forward on this. I'll make that a separate PR so we can run the build against the new test without the change in behavior (and maybe you'll think of a new test to add as well along the same lines?) and then can eventually merge those tests in this PR and test more before moving on this PR.

Reply:
Confirming, with the test in #580, `loc` gives desired behavior, `iloc` does not.